### ISSUE TITLE
Missing URL encoding in batch requests

### ISSUE THREAD

**Issue by ja-ne** [SAP/cloud-sdk-js#1226]
Hello SAP Cloud SDK,

in a previous [issue](https://github.com/SAP/cloud-sdk-js/issues/1129) I mentioned double URL encoding in batch requests (version < 1.41.0). Unfortunately, my example was incorrect. I am sorry.

**Describe the bug**
According to the [ODATA documentation](https://help.sap.com/doc/5890d27be418427993fafa6722cdc03b/Cloud/en-US/OdataV2.pdf#page=152) referred to by SAP S/4HANA Cloud APIs query operations in batch requests (e.g. retrieving entities) must "be formatted as an HTTP request just as if it was a standalone HTTP request" which includes percent encoding special characters if necessary.

**To Reproduce**
With the fix for the issue mentioned above in version 1.41.0 the following example produces a malformed syntax error due to missing URL encoding:

```
const destination = getDestinationFromSomewhere();
const request = Entity.requestBuilder()
	.getAll()
	.filter(Entity.SOMEPROPERTY.equals('Hello World'));
const result = await batch(request).execute(destination);
```

The generated batch request body is:
```
--batch_<id>
Content-Type: application/http
Content-Transfer-Encoding: binary

GET /sap/opu/odata/sap/SERVICENAME/Entity?$format=json&$filter=(SomeProperty  eq 'Hello%20World') HTTP/1.1
Content-Type: application/json
Accept: application/json


--batch_<id>

```
**Expected behavior**
URL encoding is applied once.
```
--batch_<id>
Content-Type: application/http
Content-Transfer-Encoding: binary

GET /sap/opu/odata/sap/SERVICENAME/Entity?$format=json&$filter=(SomeProperty%20eq%20'Hello%20World') HTTP/1.1
Content-Type: application/json
Accept: application/json


--batch_<id>

```

**Used Versions:**

- node version: 14.15.4
- npm version: 6.14.10
- @sap-cloud-sdk/core 1.41.0

Could you please have a look at this? Am I missing something?

Thanks in advance.

**Comment by jjtang1985** [comment]
Hi @ja-ne ,

thanks for your feedback with a nice description.
I added this item to the backlog and I assume this has the same priority as the previous one, which is medium, as some workarounds are possible.
Please let me know, if you have questions.

**Comment by ja-ne** [comment]
Hi @jjtang1985,

thanks for investigating. Medium priority is ok, I have a workaround.

**Comment by gregorwolf** [comment]
Dear @jjtang1985 ,

would be great if the priority of this issue can be increased. As we have to do complex queries that we need to combine on the client we have to avoid every roundtrip that can be avoided. You can find details of our scenario in my post: [Suggestion to reference the result of requests in subsequent requests of an OData $batch query](https://blogs.sap.com/2021/04/30/suggestion-to-reference-the-result-of-requests-in-subsequent-requests-of-an-odata-batch-query/).

Best regards
Gregor

**Comment by jjtang1985** [comment]
Hi @gregorwolf ,

we are actually working on this issue and will let you know once we have updates.

Best regards,
Junjie

**Comment by jjtang1985** [comment]
Hi all,
the fix was merged in the PR #1283, and is available if you try `canary` version.

The stable version will be released soon.
